FRENCH/src/ch10-02-traits.md

```diff
@@ -0,0 +1,1190 @@
+```
+
+<!--
+Returning either a `NewsArticle` or a `Tweet` isn’t allowed due to restrictions
+around how the `impl Trait` syntax is implemented in the compiler. We’ll cover
+how to write a function with this behavior in the [“Using Trait Objects That
+Allow for Values of Different
+Types”][using-trait-objects-that-allow-for-values-of-different-types]<!--
+ignore -- > section of Chapter 17.
+-->
+
+Retourner soit un `ArticleDePresse`, soit un `Tweet` n'est pas autorisé à cause
+des restrictions sur comment la syntaxe `impl Trait` est implémentée dans le
+compilateur. Nous allons voir comment écrire une fonction avec ce comportement
+dans une section du
+[chapitre 17][using-trait-objects-that-allow-for-values-of-different-types]<!--
+ignore -->.
+
+<!--
+### Fixing the `largest` Function with Trait Bounds
+-->
+
+### Résoudre la fonction `le_plus_grand` avec les traits liés
+
+<!--
+Now that you know how to specify the behavior you want to use using the generic
+type parameter’s bounds, let’s return to Listing 10-5 to fix the definition of
+the `largest` function that uses a generic type parameter! Last time we tried
+to run that code, we received this error:
+-->
+
+Maintenant que vous savez comment renseigner le comportement que vous souhaitez
+utiliser en utilisant les paramètres de types génériques liés, retournons à
+l'encart 10-5 pour résoudre la définition de la fonction `le_plus_grand` qui
+utilise un paramètre de type générique ! La dernière fois qu'on a essayé de
+lancer ce code, nous avions l'erreur suivante :
+
+<!--
+```text
+{{#include ../listings-sources/ch10-generic-types-traits-and-lifetimes/listing-10-05/output.txt}}
+```
+-->
+
+```text
+{{#include ../listings/ch10-generic-types-traits-and-lifetimes/listing-10-05/output.txt}}
+```
+
+<!--
+In the body of `largest` we wanted to compare two values of type `T` using the
+greater than (`>`) operator. Because that operator is defined as a default
+method on the standard library trait `std::cmp::PartialOrd`, we need to specify
+`PartialOrd` in the trait bounds for `T` so the `largest` function can work on
+slices of any type that we can compare. We don’t need to bring `PartialOrd`
+into scope because it’s in the prelude. Change the signature of `largest` to
+look like this:
+-->
+
+Dans le corps de `le_plus_grand` nous voulions comparer les deux valeurs du
+type `T` en utilisant l'opérateur *plus grand que* (`>`). Comme cet opérateur
+est défini comme une méthode par défaut dans le trait de la bibliothèque
+standard `std::cmp::PartialOrd`, nous devons préciser `PartialOrd` dans les
+traits liés pour `T` afin que la fonction `le_plus_grand` puisse fonctionner sur
+les slices de n'importe quel type que nous pouvons comparer. Nous n'avons pas
+besoin d'importer `PartialOrd` dans la portée car il est importé dans l'étape
+préliminaire. Changez la signature de `le_plus_grand` par quelque chose comme
+ceci :
```

Comment:
Not quite sure why `the prelude` is translated to `étape préliminaire`. I think `prélude` would fit better here.